app/controllers/survey_controller.rb

```diff
@@ -1,5 +1,40 @@
 class SurveyController < ApplicationController
   def index
+    @survey = Survey.new
+  end
+
+  def create
+    current_user.feelings.create(params['feeling'])
+
+    params['time'].each do |time|
+      current_user.time_preference.create("#{time}": true)
+    end
+
+    params['music'].each do |music|
+      if music && current_user.music_prefence[music]
+        current_user.music_preference.udpate("#{music}": true)
+      elsif music && !current_user.music_preference['custom']
```

Comment:
Will you explain the `!` here?

Reply:
@freeheeling  if the loop has come through and custom does not already exist, add the string to custom, otherwise, if custom is populated go to custom2